src/update_agent/actor.rs

```diff
@@ -48,16 +48,64 @@ impl Handler<LastRefresh> for UpdateAgent {
     }
 }
 
-pub(crate) struct RefreshTick {}
+/// Error thrown when the command that attempted to initiate a refresh tick
+/// is not permitted to do so in the current state of the update agent.
+#[derive(Debug, Fail)]
+struct TickPermissionError {}
```

Comment:
`Permission` usually has a different semantic nuance, related to a lack of privileges. In this case this is mostly related to an unsatisfiable request or an unfulfilled precondition. I don't a proper suggestion for this right now, but it would be better to rename this.

Reply:
`InvalidStateError`?